SimPEG/maps.py

```diff
@@ -2949,17 +2950,28 @@ class Mesh2Mesh(IdentityMap):
     """
 
     def __init__(self, meshes, indActive=None, **kwargs):
+        # Sanity checks for the meshes parameter
+        if not isinstance(meshes, Iterable):
+            raise ValueError(
```

Comment:
Should be a `TypeError` here (Incorrect type passed to the object)

Reply:
Also, checking if it is an instance of `Iterable` does not always succeed (The documentation states it does not work if the container iterates with a `__getitem__` call.) I would leave it as is, but specifically catch the TypeError that is thrown if the input can't be unpacked.

Reply:
Right. I fell back to the original version but now we are catching the exception. I also removed the extra check for the size of the `meshes`, since we would catch that in the `try`.